rocks-lib/src/manifest/metadata.rs

```diff
@@ -42,19 +48,71 @@ impl ManifestMetadata {
         self.repository.contains_key(rock_name)
     }
 
-    pub fn available_versions(&self, rock_name: &PackageName) -> Option<Vec<&String>> {
+    pub fn available_versions(&self, rock_name: &PackageName) -> Option<Vec<&PackageVersion>> {
         if !self.has_rock(rock_name) {
             return None;
         }
 
         Some(self.repository[rock_name].keys().collect())
     }
 
-    pub fn latest_version(&self, rock_name: &PackageName) -> Option<&String> {
+    pub fn latest_version(&self, rock_name: &PackageName) -> Option<&PackageVersion> {
         if !self.has_rock(rock_name) {
             return None;
         }
 
         self.repository[rock_name].keys().sorted().last()
     }
+
+    pub fn latest_match(&self, lua_package_req: &LuaPackageReq) -> Option<LuaPackage> {
+        if !self.has_rock(lua_package_req.name()) {
+            return None;
+        }
+
+        let version = self.repository[lua_package_req.name()]
+            .keys()
+            .filter(|version| lua_package_req.version_req().matches(version))
+            .sorted()
+            .last()?;
+
+        Some(LuaPackage::new(
+            lua_package_req.name().to_owned(),
+            version.to_owned(),
+        ))
+    }
+}
+
+#[derive(serde::Deserialize)]
+pub struct ManifestRockEntry {
+    /// e.g. "linux-x86_64", "rockspec", "src", ...
+    pub arch: String,
```

Comment:
question: we'd probably benefit from making this an enum too, wouldn't we? Should this be given a TODO for the future? Note that the `Triple` type can be used to represent a "linux-arch" string quite nicely (with serialization and such).

Reply:
Created #62